src/java/org/apache/cassandra/sensors/RequestSensors.java

```diff
@@ -110,6 +110,29 @@ public void syncAllSensors()
         }
     }
 
+    /**
+     * Avoids using additional ConcurrentHashmap as that turns out to be
+     * a performance bottlneck (in particular hashCode operations on Sensor were taking a lots of CPU cycles)
+     */
+    public void syncAllSensorsNew()
```

Comment:
I don't think this is equivalent. Do tests pass when you substitute the original method with this one ?

Reply:
Yep, `RequestSensorsTest` passes just fine

Reply:
I see. Cool!